src/structures/CommandInteraction.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+const APIMessage = require('./APIMessage');
+const Interaction = require('./Interaction');
+const Snowflake = require('../util/Snowflake');
+
+/**
+ * Represents a command interaction, see {@link InteractionClient}.
+ * @extends {Interaction}
+ */
+class CommandInteraction extends Interaction {
```

Comment:
In my opinion, it should extends the `ApplicationCommand` (who could be partial), because of consistency, and `Interaction` would have the command property.

Reply:
`Interaction` has very different properties than `ApplicationCommand`, that `CommandInteraction` needs to function. Furthermore, if you look at the data that is provided in an Interaction event, the data about the command is not the same as an ApplicationCommand.
One possible thought is to have a `CommandInteraction#command` that is the `ApplicationCommand`, however that breaks down immediately when you realize that your commands aren't being cached in a manager (which there is no need for).

And to finish it off, there will be future types of Interactions that need an `Interaction` as well.

Reply:
It seems that you misunderstood me ; I would like to have `CommandInteraction#command`, to avoid confusing the source of the interaction and the interaction itself.
Also, imo a manager would be great, as discord does not send the command data when receiving an interaction, and it would be nice to access it without fetching it each time.
I know that there will be several Interactions types, then we need to create a specific class to implement each of them, but I don't feel confortable with the choices made here. I feel like we're on v11 with such properties.

Reply:
I see no need for the initial `ApplicationCommand` structure when dealing with the interaction (thus no manager). However, if discord implements the suggestion to send the option type of subcommands and subcommand groups when sending an interaction, I could see there being a `CommandInteraction#command` that is a new  partial `ApplicationCommand` (would be missing guild id (since we have no way of knowing if it is a guild only command), created at, created timestamp)

Reply:
Indeed, we could regroup ApplicationCommand and CommandInteraction in this case.